src/jbuild.ml

```diff
@@ -901,16 +895,20 @@ module Library = struct
          match name, public with
          | Some n, _ -> n
          | None, Some { name = (_loc, name) ; _ }  ->
-           if dune_version >= (1, 1) then
-             name
-           else
-             of_sexp_error loc "name field cannot be omitted before version \
-                                1.1 of the dune language"
+           begin
+             match Syntax.Version.compare dune_version (1, 1) with
+             | Gt | Eq ->
```

Comment:
IMO, it would be nicer to keep using `>=` here. E.g.

```
Syntax.Version.(dune_version >= (1, 1))
```

Where `>=` would be monomorphic. But I realize this will take a little more work and possibly some functor writing these comparison helpers.

Reply:
I find using the `>=` operator clearer as well. Using a local open a bit dodgy, but this still looks better to me that the pattern matching: `Syntax.Version.(>=) dune_version (1, 1)`